PhysicsTools/PatAlgos/plugins/GenJetFlavourInfoPreserver.cc

```diff
@@ -0,0 +1,88 @@
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/Common/interface/Association.h"
+#include "DataFormats/Common/interface/RefToPtr.h"
+
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/JetReco/interface/JetCollection.h"
+#include "SimDataFormats/JetMatching/interface/JetFlavourInfo.h"
+#include "SimDataFormats/JetMatching/interface/JetFlavourInfoMatching.h"
+
+#include "DataFormats/JetReco/interface/GenJet.h"
+#include "DataFormats/JetReco/interface/GenJetCollection.h"
+
+
+namespace pat {
+
+  class GenJetFlavourInfoPreserver : public edm::stream::EDProducer<> {
+  public:
+    explicit GenJetFlavourInfoPreserver(const edm::ParameterSet & iConfig);
+    virtual ~GenJetFlavourInfoPreserver() { }
+    
+    virtual void produce(edm::Event & iEvent, const edm::EventSetup & iSetup) override;
+    
+  private:
+    const edm::EDGetTokenT<edm::View<reco::GenJet> > genJetsToken_;
+    const edm::EDGetTokenT<edm::View<reco::Jet> > slimmedGenJetsToken_;
+    
+    const edm::EDGetTokenT<reco::JetFlavourInfoMatchingCollection> genJetFlavourInfosToken_;
+    const edm::EDGetTokenT<edm::Association<std::vector<reco::GenJet> > > slimmedGenJetAssociationToken_;
+  };
+
+} // namespace
+
+pat::GenJetFlavourInfoPreserver::GenJetFlavourInfoPreserver(const edm::ParameterSet & iConfig) :
+    genJetsToken_(consumes<edm::View<reco::GenJet> >(iConfig.getParameter<edm::InputTag>("genJets"))),
+    slimmedGenJetsToken_(consumes<edm::View<reco::Jet> >(iConfig.getParameter<edm::InputTag>("slimmedGenJets"))),
+    genJetFlavourInfosToken_(consumes<reco::JetFlavourInfoMatchingCollection>(iConfig.getParameter<edm::InputTag>("genJetFlavourInfos"))),
+    slimmedGenJetAssociationToken_(consumes<edm::Association<std::vector<reco::GenJet> > >(iConfig.getParameter<edm::InputTag>("slimmedGenJetAssociation")))
+{
+    produces<reco::JetFlavourInfoMatchingCollection>();
+}
+
+void 
+pat::GenJetFlavourInfoPreserver::produce(edm::Event & iEvent, const edm::EventSetup & iSetup) {
+    using namespace edm;
+    using namespace std;
+
+    Handle<View<reco::GenJet> >      genJets;
+    iEvent.getByToken(genJetsToken_, genJets);
+
+    Handle<View<reco::Jet> >      slimmedGenJets;
+    iEvent.getByToken(slimmedGenJetsToken_, slimmedGenJets);
+
+    Handle<reco::JetFlavourInfoMatchingCollection> genJetFlavourInfos;
+    iEvent.getByToken(genJetFlavourInfosToken_, genJetFlavourInfos);
+
+    Handle<edm::Association<std::vector<reco::GenJet> > > slimmedGenJetAssociation;
+    iEvent.getByToken(slimmedGenJetAssociationToken_, slimmedGenJetAssociation);
+
+    auto jetFlavourInfos = std::make_unique<reco::JetFlavourInfoMatchingCollection>(reco::JetRefBaseProd(slimmedGenJets));
+    assert(genJets->size() == genJetFlavourInfos->size());
+
+    edm::Ref<std::vector<reco::GenJet> > slimmedGenJetRef;
+
+    int i = -1;
+    for (auto const& genJet : *genJets){
```

Comment:
hi @Andrej-CMS @slava77 @perrotta - this brings a warning as genJet is not actually used. Please change to a loop over i if genJets->refAt(i) is not the same as genJet in the loop below

Reply:
Hello @davidlange6 ,

I just pushed the changes to the initial branch on my repository: https://github.com/Andrej-CMS/cmssw/tree/JetFlavourMiniAOD92X

Since the changes here are already merged, do I need to make a new PR?

Kind regards,
Andrej